File: src/main/java/pe/nanamochi/banchus/entities/commons/LeaderboardType.java
package pe.nanamochi.banchus.entities.commons;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum LeaderboardType {
  LOCAL(0),
  GLOBAL(1),
  MODS(2),
  FRIENDS(3),
  COUNTRY(4);

  private final int value;

  public static LeaderboardType fromValue(int value) {
    for (LeaderboardType type : LeaderboardType.values()) {
      if (type.getValue() == value) {
        return type;
      }
    }
    throw new IllegalArgumentException("Invalid LeaderboardType value: " + value);
  }
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/domain/enums/LeaderboardType.kt
package pe.nanamochi.banchus.domain.enums

enum class LeaderboardType(val value: Int) {
    LOCAL(0),
    GLOBAL(1),
    MODS(2),
    FRIENDS(3),
    COUNTRY(4);

    companion object {
        fun fromValue(value: Int): LeaderboardType {
            return entries.find { it.value == value } ?: GLOBAL
        }
    }
}
